tests/unit/accounts/test_views_db.py

```diff
@@ -0,0 +1,65 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import datetime
+
+import pretend
+
+from pyramid.request import Request
+
+from warehouse.accounts import ITokenService, IUserService
+from warehouse.accounts.models import User
+from warehouse.accounts.services import DatabaseUserService, TokenService
+from warehouse.accounts.views import two_factor_and_totp_validate
+
+
+class TestTwoFactor:
+    def test_two_factor_and_totp_validate_redirect_to_account_login(
```

Comment:
this test utilises the database.

It saves the User within the database to make sure it's stored there, expunged, so we do not have it loaded in a session with values before creation (it'd use naive DateTime instead of the value converted during reading)

Then the view actually loads the user (we use both token and user services, not stubs).

This triggers the exact failed line, comparing the timestamp from token_service against the user.last_login read from the database.

I could actually call `_get_two_factor_data` directly in the test instead of the whole view, but ideally, that would mean I'd need to create a subtype to TokenInvalid exception to make sure I got the exception after the desired code path 🤔

Reply:
Ah, great. Could we just move it into https://github.com/pypa/warehouse/blob/main/tests/unit/accounts/test_views.py then?

Reply:
> I could actually call _get_two_factor_data directly in the test instead of the whole view

Since we don't _generally_ test `_`-prefixed functions, I think the current test is fine.

Reply:
Sure, I'll move it. 

Although I noticed that all tests there are stubbed, so never thought it would be a good idea to mix them (also, I personally never liked massive test files)